runtime/compiler/control/MethodToBeCompiled.hpp

```diff
@@ -144,11 +146,18 @@ struct TR_MethodToBeCompiled
    uint8_t                _weight; // Up to 256 levels of weight
    bool                   _hasIncrementedNumCompThreadsCompilingHotterMethods;
    uint8_t                _jitStateWhenQueued;
+
 #if defined(J9VM_OPT_JITSERVER)
-   bool                   _remoteCompReq; // Comp request should be sent remotely to JITServer
-   JITServer::ServerStream  *_stream; // A non-NULL field denotes an out-of-process compilation request
-   TR_Hotness             _origOptLevel; //  Cache original optLevel when transforming a remote sync compilation to a local cheap one
-   bool                   _shouldUpgradeOutOfProcessCompilation; // Flag used to determine whether a cold local compilation should be upgraded by LPQ
+   bool                     _remoteCompReq; // Comp request should be sent remotely to JITServer
+   JITServer::ServerStream *_stream; // A non-NULL field denotes an out-of-process compilation request
+   TR_Hotness               _origOptLevel; // Cache original optLevel when transforming a
+                                           // remote sync compilation to a local cheap one
+   bool                     _shouldUpgradeOutOfProcessCompilation; // Flag used to determine whether a cold local
+                                                                   // compilation should be upgraded by LPQ
+   bool                     _aotCacheStore; // True if the result of this out-of-process compilation will be
+                                            // stored in JITServer AOT cache; always false at the client
+   uint32_t                 _methodIndex; // Index in the array of methods of the defining class
+   const AOTCacheClassChainRecord *_definingClassChainRecord;// Only used at the JITServer
```

Comment:
Can we store these fields in `CompilationInfoPerThreadRemote` instead of `TR_MethodToBeCompiled`? The reason is that in some cases we can have thousands of method entries in the queue, but only tens of compilation threads. If, I would prefer a separate commit with this change.

Reply:
Yes, I can move these to `compInfoPT`. If memory footprint of this structure is a concern, its size can be reduced by up to 16 bytes by storing boolean flags as bits and reordering some fields (maybe in a future PR).